session_0/word_processing.rb

```diff
@@ -2,13 +2,22 @@
 # any non-lower case alphabet (A..Z) to corresponding lower case
 # alphabet
 def lower_case(words)
-  raise NotImplementedError # TODO
+  #raise NotImplementedError # TODO
+  sub = []
+  words.each do |i|
```

Comment:
Can you use map function in this function too?

Reply:
I was under the impression that the map function modifies the original array, that's why I used the each function.

Reply:
read up on the difference between map and map! functions
https://softwareengineering.stackexchange.com/questions/228284/auxiliary-space-complexity-of-map-vs-map

Reply:
> read up on the difference between map and map! functions
> https://softwareengineering.stackexchange.com/questions/228284/auxiliary-space-complexity-of-map-vs-map

Thanks, I read up on the differences between each, map and map! only after I submitted my code 😅